core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/provision/service/ClusterProxyService.java

```diff
@@ -115,23 +118,42 @@ public void deregisterCluster(Stack stack) {
 
     private ConfigRegistrationRequest createProxyConfigRequest(Stack stack) {
         ConfigRegistrationRequestBuilder requestBuilder = new ConfigRegistrationRequestBuilder(stack.getResourceCrn())
-                .withAliases(singletonList(clusterId(stack.getCluster()))).withServices(serviceConfigs(stack));
+                .withAliases(singletonList(clusterId(stack.getCluster())))
+                .withServices(serviceConfigs(stack));
         if (stack.getTunnel().useCcmV1()) {
-            requestBuilder.withAccountId(getAccountId(stack)).withTunnelEntries(tunnelEntries(stack));
-        } else if (stack.getTunnel().useCcmV2OrJumpgate()) {
-            requestBuilder.withAccountId(getAccountId(stack)).withServices(serviceConfigsForCcmV2(stack)).withCcmV2Entries(ccmV2Configs(stack));
+            requestBuilder.withAccountId(getAccountId(stack))
+                    .withTunnelEntries(tunnelEntries(stack));
+        } else if (stack.getTunnel().useCcmV2()) {
+            requestBuilder.withAccountId(getAccountId(stack))
+                    .withServices(serviceConfigsForCcmV2(stack))
+                    .withCcmV2Entries(ccmV2Configs(stack));
+        } else if (stack.getTunnel().useCcmV2Jumpgate()) {
+            requestBuilder.withEnvironmentCrn(stack.getEnvironmentCrn())
```

Comment:
while I see the env crn here, I miss it on the FreeIPA side. I thought I would see it during FreeIPA CCMv2 registration in case it's a jumpgate type, or always

Reply:
I discussed with @zabhi  and he suggested the way it is coded now. That is: freeipa registration **should not have** the environmentCrn but has the CCMv2config and isCCMv2 enabled. The dl/dh registration **should have** the environmentCrn but no CCMv2Config. The isCCMv2 flag must be enabled too. This is also tested by unit tests.

Reply:
ok, I looked it up in the code, so actually CCMv2 registration happens on FreeIPA with an env crn, while it happens without it on CB side. So it must be how it's connected.

Reply:
vica versa but yes. freeipa without env crn, CB core with env crn

Reply:
I was talking about CCMv2 reg, you are talking about CP reg. but both of us is right :)

Reply:
Yes right :) I have not checked the CCMv2 inverting proxy agent registration so that is unchanged and works as you have described.